### PR TITLE
Composite node

### PR DESCRIPTION
**Pull request by FHell**
I have implemented a CompositeNode Type that can aggregate current sources. This is a first step towards a Composite Node that can hold up to one voltage, arbitrary power and arbitrary current sources.

### CODE REVIEW

**Comment by FHell** [comment]
I added comments that sketch the full design of a CompositeNode type.

File: src/nodes/CompositeNode.jl [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+        # The key implementation detail is that we stack the internal variables
+        # of all nodes in the order I, P and U. The array of indices makes sure
+        # that these are combined correctly with the first two indices that represent
+        # the voltage or the current constraints.
+
+        if CurrentNodes != nothing
+            mix *= "I"
+            for current_node in CurrentNodes
+                append!(symbols, symbolsof(current_node)[3:end])
```
Comment (SabineAuer) [inline src/nodes/CompositeNode.jl]: what if a current node has no additional internal variables? like the CSIMinimal?

Comment (FHell) [reply src/nodes/CompositeNode.jl]: Then this append! does nothing.
```julia
a = [1,2]
b = [6,7]
append!(a, b[3:2])
append!(a, b[3:end])
a == [1,2] # true
```

**Comment by SabineAuer** [comment]
@FHell: did you implement an example test system with the composite node type? I tried to create a power grid with a node list consisting of a composite node, a slack bus and a PQ bus and disturb the PQ bus with a power perturbation. Then it finds a fixed point for the systems but the simulation is not running through (I had to stop it after a long time).

**Comment by SabineAuer** [comment]
@luap-pik: I added the runexample.jl I was talking about. Can you check why the simulation is not converging/ running through?

**Comment by luap-pik** [comment]
In `run_example.jl` the function `rand_real` is not available, you can use `rand`.
However, I suggest to rebase on master before debugging further since we the branch
seems to be based on an older version?

**Comment by luap-pik** [comment]
I fixed some lines, now simulate gives a valid result (though unstable, probably due to the random parameters).

In the definition, I changed it to `struct CompositeNode{T} <: AbstractNode ` which should be more appropriate. I don see why we need a mutable struct here.

**Pull request merged**